Sources/SwiftfulUI/ViewModifiers/OnAppFirstRunModifier.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  OnAppFirstRunModifier.swift
+//
+//
+//  Created by Ricky Stone on 17/11/2023.
+//
+
+import SwiftUI
+
+struct OnAppFirstRunModifier: ViewModifier {
+    let action: @MainActor () -> Void
+
+    func body(content: Content) -> some View {
+        content
+            .onAppear {
+                if UserDefaults.standard.bool(forKey: "appHasBeenLaunchedOnce") == false {
+                    action()
+                    UserDefaults.standard.set(true, forKey: "appHasBeenLaunchedOnce")
+                }
+            }
+    }
+}
```

Comment:
This is a little confusing as a ViewModifier because there's nothing stopping the developer from adding more than one **onAppFirstRun** modifier. In that situation, only the first one should execute.